quartz/components/scripts/graph.inline.ts

```diff
@@ -42,20 +42,38 @@ async function renderGraph(container: string, fullSlug: FullSlug) {
     linkDistance,
     fontSize,
     opacityScale,
+    removeTags,
+    showTags,
   } = JSON.parse(graph.dataset["cfg"]!)
 
   const data = await fetchData
 
   const links: LinkData[] = []
+  const tags: SimpleSlug[] = []
+
   const validLinks = new Set(Object.keys(data).map((slug) => simplifySlug(slug as FullSlug)))
+
   for (const [src, details] of Object.entries<ContentDetails>(data)) {
     const source = simplifySlug(src as FullSlug)
     const outgoing = details.links ?? []
+
     for (const dest of outgoing) {
       if (validLinks.has(dest)) {
         links.push({ source, target: dest })
       }
     }
+
+    if (showTags) {
+      const localTags = details.tags
+        .filter((tag) => !removeTags.includes(tag))
+        .map((tag) => simplifySlug(("tags/" + tag) as FullSlug))
+
+      tags.push(...localTags.filter((tag) => !tags.includes(tag)))
+
+      for (const tag of localTags) {
+        links.push({ source, target: tag })
```

Comment:
tag nodes don't always exist, you may need to manually construct and insert them here too. they only exist for tags where you've added a description for it

Reply:
I don't quite understand?

Reply:
This code adds the _link_ to a node for the tag but that node isn't guaranteed to exist. If you make a `tags/something.md` file then the `something` tag while have a node but another tag like `asdf` won't have one unless `tags/asdf.md` exists. You should create a node in L101 for all tags if it doesn't already exist

Reply:
Ah, I just insert the tags into the neighborhood on L97